### PR TITLE
Use GitHub workflows instead of travis.org for CI and publishing.

### PR DESCRIPTION
**Pull request by mkruisselbrink**

### CODE REVIEW

**Comment by mkruisselbrink** [comment]
This doesn't currently do w3c publishing yet, since afaik the w3c spec-prod github action doesn't support that for bikeshed specs yet. Since we probably don't want to lose that, we should hold of on merging this...

**Comment by annevk** [comment]
Maybe @sideshowbarker can help with this as he recently converted a bunch of repos.

File: .github/workflows/pr-push.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  push:
+    branches: [main]
+
+jobs:
+  main:
+    name: Build, Validate and Deploy
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: sidvishnoi/spec-prod@v1
```
Comment (sideshowbarker) [inline .github/workflows/pr-push.yml:13]:
```suggestion
      - uses: w3c/spec-prod@v2
```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]
With the suggested commit applied, this should work as-is

**Review by sideshowbarker — APPROVED** [review]

**Comment by mkruisselbrink** [comment]
I think this is still missing the auto-publishing as WD? Or does that not need some configuration as well?

**Comment by sideshowbarker** [comment]
> I think this is still missing the auto-publishing as WD? Or does that not need some configuration as well?

Yeah it needs a couple more lines in the config file. But I’ve now added them, in https://github.com/w3c/FileAPI/pull/167.

**Pull request merged**